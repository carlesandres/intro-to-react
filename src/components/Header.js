import React from 'react';

class Header extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      likes: 0
    };

    this.incrementLikes = this.incrementLikes.bind(this);
  }

  componentDidMount() {
    // We are increasing the number of likes by 1 every second
    setInterval( () => {
      const newState = this.state.likes + 1;
      this.setState({ likes: newState });
    }, 5000);
  }

  incrementLikes(event) {
    this.setState({ likes: this.state.likes + 1 });
  }

  render() {
    return (
      <div className="header">
        <h1>{this.props.title || 'Finantial Times'}</h1>
        <h2>{this.props.subtitle || ''}</h2>
        <div>Likes: {this.state.likes}</div>
        <button onClick={this.incrementLikes}>Like</button>
      </div>
    );
  }
}

export default Header;
