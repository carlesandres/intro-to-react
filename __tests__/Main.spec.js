import React from 'react';
import Header from '../src/components/Header';
import renderer from 'react-test-renderer';

test('Header renders as expect with no params', () => {
  const component = renderer.create(
    <Header />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
