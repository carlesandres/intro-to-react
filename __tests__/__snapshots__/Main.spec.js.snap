// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expect with no params 1`] = `
<div
  className="header"
>
  <h1>
    Finantial Times
  </h1>
  <h2>
    
  </h2>
  <button
    className="switch"
  >
    Switch Layout
  </button>
</div>
`;
